src/main/java/io/cdap/plugin/http/source/common/HttpErrorHandlingStrategy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.http.source.common;
+
+/**
+ * Indicates error handling strategy which will be used to handle unexpected http status codes.
+ */
+public enum HttpErrorHandlingStrategy implements EnumWithValue {
+  SUCCESS("Success"),
+  FAIL("Fail"),
+  SKIP("Skip"),
+  SEND_TO_ERROR("Send to error"),
+  RETRY_AND_FAIL("Retry and fail"),
```

Comment:
Now that I think of this more, I don't think we need the 'retry and xyz' strategies. Retries and failure handling can be orthogonal. We can just have Fail, Skip, Send to Error. If no retry strategy is given, no retries will be done. Otherwise the error strategy only applies after retries have been exhausted.

Reply:
My original though that something like 500 error code. Might require retries. While something like 400 will not, since it's client error.

Reply:
Hm I see, that makes sense. Let's keep it how you have it then.